Include cert-manager for local deployment

baijum opened the pull request:
Signed-off-by: Baiju Muthukadan <baiju.m.mail@gmail.com>

baijum commented on README.md:
````diff
@@ -101,7 +101,7 @@ You’ll need a Kubernetes cluster to run against. You can use [KIND](https://si
 
    ```sh
-   export KO_DOCKER_REPO=kind.local/servicebinding
+   export KO_DOCKER_REPO=kind.local
````
Looks like the expected kind cluster is the default one with the name `kind`.

scothis reviewed (changes requested):
We cannot automatically install cert-manager as it will inherently conflict with any existing install of cert-manager that happens to be in the cluster. We can have independent make targets to install/uninstall cert-manager, we just can't depend on them in any of the other targets.

baijum commented on Makefile:
```diff
@@ -74,9 +74,13 @@ test: manifests generate fmt vet ## Run tests.
 
+.PHONY: cert-manager
+cert-manager:
+	$(KAPP) deploy -a $(KAPP_APP) --wait-timeout 5m -y -f https://github.com/cert-manager/cert-manager/releases/download/v1.8.0/cert-manager.yaml
```
Looks like the cert-manager is deployed in the `cert-manager` namespace. But that's not explicitly mentioned here. Let me know if I should specify the namespace as `cert-manager`.

scothis replied:
Namespaces for apps is a bit chicken<->egg. You cannot create the app in a namespace that doesn't exist. If you create the namespace to create the app within it, and the manifest contains that same namespace, it will fail because you're asking kapp to own a namespaces that already exists.

I usually end up being lazy and create everything in the default namespace for dev cluster. For important clusters, I'll create a namespace dedicated to holding these app references. The resource is a ConfigMap, so name collisions are possible.

baijum commented:
> We cannot automatically install cert-manager as it will inherently conflict with any existing install of cert-manager that happens to be in the cluster. We can have independent make targets to install/uninstall cert-manager, we just can't depend on them in any of the other targets.

I have removed the dependency. Also added a note about the make target in the README.

scothis commented on Makefile:
```diff
@@ -77,6 +77,14 @@ test: manifests generate fmt vet ## Run tests.
 deploy: manifests ## Deploy controller to the K8s cluster specified in ~/.kube/config.
 	$(KAPP) deploy -a $(KAPP_APP) -n $(KAPP_APP_NAMESPACE) -c -f config/kapp -f <($(KO) resolve --platform $(KO_PLATFORMS) -f config/servicebinding-runtime.yaml)
 
+.PHONY: deploy-cert-manager
+deploy-cert-manager: ## Deploy cert-manager to the K8s cluster specified in ~/.kube/config.
+	$(KAPP) deploy -a cert-manager --wait-timeout 5m -c -f https://github.com/cert-manager/cert-manager/releases/download/v1.8.0/cert-manager.yaml
+
+.PHONY: undeploy-cert-manager
+undeploy-cert-manager: ## Undeploy cert-manager from the K8s cluster specified in ~/.kube/config.
+	$(KAPP) delete -a cert-manager
+
```
We can reuse the namespace, which is "default" by default

```suggestion
.PHONY: deploy-cert-manager
deploy-cert-manager: ## Deploy cert-manager to the K8s cluster specified in ~/.kube/config.
	$(KAPP) deploy -a cert-manager -n $(KAPP_APP_NAMESPACE) --wait-timeout 5m -c -f https://github.com/cert-manager/cert-manager/releases/download/v1.8.0/cert-manager.yaml

.PHONY: undeploy-cert-manager
undeploy-cert-manager: ## Undeploy cert-manager from the K8s cluster specified in ~/.kube/config.
	$(KAPP) delete -a cert-manager -n $(KAPP_APP_NAMESPACE)

```
